File: Taller de Haskell/Problema 2/problema2.py
# Contraseña
contraseña_guardada = "1234"

# Se le pide al usuario que escriba la contraseña
contraseña_usuario = input("Escribe la contraseña: ")

# Se convierten ambas cadenas a minúsculas para compararlas
if contraseña_usuario.lower() == contraseña_guardada.lower():
    print("La contraseña es correcta")
else:
    print("La contraseña es incorrecta")

File: Taller de Haskell/Problema 2/problema2.hs
import Data.Char (toLower)

-- Esta es la función que convierte una cadena a minúsculas
aMinus :: String -> String
aMinus texto = map toLower texto

-- Esta es la función que compara dos contraseñas ignorando las mayúsculas
verificarContraseña :: String -> String -> Bool
verificarContraseña almacenada entrada =
    aMinus almacenada == aMinus entrada
    
main :: IO ()
main = do
    putStrLn "Escribe la contraseña: "
    entrada <- getLine
    let contraseñaGuardada = "1234"
    if verificarContraseña contraseñaGuardada entrada
        then putStrLn "La contraseña es correcta"
        else putStrLn "La contraseña es incorrecta"
